build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -217,8 +225,12 @@ private List<CompilationDatabaseInformation> populateObjList(IProject project, I
 						outputLocation + "", inputStrings, sourceLocation, outputLocation); //$NON-NLS-1$
 
 				IBuildMacroProvider provider = ManagedBuildManager.getBuildMacroProvider();
-				String resolvedOptionFileContents = provider.resolveValueToMakefileFormat(cmdLInfo.getCommandLine(), "", //$NON-NLS-1$
-						" ", IBuildMacroProvider.CONTEXT_FILE, //$NON-NLS-1$
+				String compilerName = CompilationDatabaseGenerator.getCompilerName(tool);
+				String commandLine = cmdLInfo.getCommandLine();
+				commandLine = commandLine.replace(compilerName, "").trim(); //$NON-NLS-1$
+				commandLine = commandLine.replaceAll("\\s+", " ").trim(); //$NON-NLS-1$ //$NON-NLS-2$
+				String resolvedOptionFileContents = provider.resolveValueToMakefileFormat(
+						findCompilerInPath(tool, config) + " " + commandLine, "", " ", IBuildMacroProvider.CONTEXT_FILE, //$NON-NLS-1$//$NON-NLS-2$ //$NON-NLS-3$
```

Comment:
findCompilerInPath can return `null` - so I think we have to handle that either by having it not return null, or if it does return null than don't modify the command line? 

With the null AFAICT you can end up with a string like `null -c a.c -o a.o` in the compile_commands.json